Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -766,6 +771,7 @@ namespace Babylon
 
             arcana::trace_region endFrameRegion{"NativeXR::EndFrame"};
 
+            m_sessionState->Frame->Render();
             m_sessionState->Frame.reset();
```

Comment:
There is something I'm still not following... the current frame is updated on the JS thread, and things like anchor creation also happen on the JS thread. So do we really need a shared_ptr, or do we just need to extend the lifetime of the unique_ptr beyond `EndFrame`? Since you have an explicit `Render` function now, couldn't you just skip the call to `.reset()` and allow it to only reset/release when the current frame is replaced by the next frame?

Reply:
But it also means there will be 1 more frame latency. is this acceptable?

Reply:
If you keep your `Render` function but remove the `reset`, then it doesn't seem like it would result in a 1 frame latency.

Regarding the lifetime though, I guess the JS side can hold the JS XRFrame as long as it wants, and it should be able to create an anchor from an "old" frame, is that right? Is that how it works in the browser?